cmake/kokkos_arch.cmake

```diff
@@ -301,6 +302,14 @@ IF (KOKKOS_ARCH_A64FX)
   )
 ENDIF()
 
+IF (KOKKOS_ARCH_ARMV9_GRACE)
+  SET(KOKKOS_ARCH_ARM_NEON ON)
+  COMPILER_SPECIFIC_FLAGS(
+    COMPILER_ID KOKKOS_CXX_HOST_COMPILER_ID
+    DEFAULT -mcpu=neoverse-v2  -msve-vector-bits=128
```

Comment:
`neoverse-v2` is not a valid `gcc-12.2` architecture, do we need to protect this flag?

Reply:
Looks like we should protect it.

Reply:
It should be protected now for CMake.